From 1b4944795cb99a3862576c2a3da5ebe96cadd635 Mon Sep 17 00:00:00 2001
From: Mashopy <eliasgheeraert@gmail.com>
Date: Mon, 15 Dec 2025 08:13:46 +0100
Subject: [PATCH 3/3] ImsService: Remove references to TelephonyMetrics

Google deprecated TelephonyMetrics class in Android 16 QPR2.
Since it's just used for metrics purposes, we can safely remove its references.

Change-Id: I00bf6ade3d62d9d9436ebca68709453929134fd2
---
 .../ims/ril/ImsRILAdapter$RilHandler.smali    | 20 -------
 .../com/mediatek/ims/ril/ImsRILAdapter.smali  | 58 -------------------
 .../mediatek/wfo/ril/MwiRIL$RilHandler.smali  | 25 --------
 smali/com/mediatek/wfo/ril/MwiRIL.smali       | 36 +-----------
 4 files changed, 1 insertion(+), 138 deletions(-)

diff --git a/smali/com/mediatek/ims/ril/ImsRILAdapter$RilHandler.smali b/smali/com/mediatek/ims/ril/ImsRILAdapter$RilHandler.smali
index fae19c3..847f707 100644
--- a/smali/com/mediatek/ims/ril/ImsRILAdapter$RilHandler.smali
+++ b/smali/com/mediatek/ims/ril/ImsRILAdapter$RilHandler.smali
@@ -1054,24 +1054,6 @@
     .line 697
     iget-object v3, p0, Lcom/mediatek/ims/ril/ImsRILAdapter$RilHandler;->this$0:Lcom/mediatek/ims/ril/ImsRILAdapter;
 
-    invoke-static {v3}, Lcom/mediatek/ims/ril/ImsRILAdapter;->access$300(Lcom/mediatek/ims/ril/ImsRILAdapter;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v3
-
-    iget-object v4, p0, Lcom/mediatek/ims/ril/ImsRILAdapter$RilHandler;->this$0:Lcom/mediatek/ims/ril/ImsRILAdapter;
-
-    iget-object v4, v4, Lcom/mediatek/ims/ril/ImsRILAdapter;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v4}, Ljava/lang/Integer;->intValue()I
-
-    move-result v4
-
-    iget v5, v1, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
-
-    iget v6, v1, Lcom/mediatek/ims/ril/RILRequest;->mRequest:I
-
-    invoke-virtual {v3, v4, v5, v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilTimeoutResponse(III)V
-
     .line 700
     .end local v2    # "timeoutResponse":Ljava/lang/Object;
     :cond_6
@@ -1294,8 +1276,6 @@
     :goto_2
     return-void
 
-    nop
-
     :pswitch_data_0
     .packed-switch 0x2
         :pswitch_a
diff --git a/smali/com/mediatek/ims/ril/ImsRILAdapter.smali b/smali/com/mediatek/ims/ril/ImsRILAdapter.smali
index c88fdea..77887dd 100644
--- a/smali/com/mediatek/ims/ril/ImsRILAdapter.smali
+++ b/smali/com/mediatek/ims/ril/ImsRILAdapter.smali
@@ -188,8 +188,6 @@
 
 .field mLastNITZTimeInfo:[Ljava/lang/Object;
 
-.field private mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
 .field public mModemIndication:Lcom/mediatek/ims/ril/ModemIndication;
 
 .field public mModemResponse:Lcom/mediatek/ims/ril/ModemResponse;
@@ -600,13 +598,6 @@
 
     iput-object v0, v1, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMtkRadioVersion:Ljava/util/Map;
 
-    .line 434
-    invoke-static {}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->getInstance()Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v0
-
-    iput-object v0, v1, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
     .line 441
     new-instance v0, Lcom/mediatek/ims/ril/ImsRILAdapter$DtmfQueueHandler;
 
@@ -1668,16 +1659,6 @@
     return-object v0
 .end method
 
-.method static synthetic access$300(Lcom/mediatek/ims/ril/ImsRILAdapter;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-    .locals 1
-    .param p0, "x0"    # Lcom/mediatek/ims/ril/ImsRILAdapter;
-
-    .line 319
-    iget-object v0, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    return-object v0
-.end method
-
 .method static synthetic access$400(Lcom/mediatek/ims/ril/ImsRILAdapter;Lcom/mediatek/ims/ril/RILRequest;)V
     .locals 0
     .param p0, "x0"    # Lcom/mediatek/ims/ril/ImsRILAdapter;
@@ -5597,23 +5578,6 @@
     .line 4420
     :cond_1
     :goto_0
-    iget-object v1, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget-object v0, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v0}, Ljava/lang/Integer;->intValue()I
-
-    move-result v2
-
-    iget v3, p1, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
-
-    iget v5, p1, Lcom/mediatek/ims/ril/RILRequest;->mRequest:I
-
-    move v4, p2
-
-    move-object v6, p4
-
-    invoke-virtual/range {v1 .. v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilSolicitedResponse(IIIILjava/lang/Object;)V
 
     .line 4423
     iget v0, p1, Lcom/mediatek/ims/ril/RILRequest;->mRequest:I
@@ -8612,18 +8576,6 @@
 
     invoke-virtual {v0, v2}, Lcom/mediatek/ims/ril/RadioVoiceProxy;->acceptCall(I)V
 
-    .line 1703
-    iget-object v2, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget-object v3, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v3}, Ljava/lang/Integer;->intValue()I
-
-    move-result v3
-
-    iget v4, v1, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
-
-    invoke-virtual {v2, v3, v4}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeRilAnswer(II)V
     :try_end_0
     .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
     .catch Ljava/lang/RuntimeException; {:try_start_0 .. :try_end_0} :catch_0
@@ -17792,14 +17744,6 @@
 
     .line 4559
     :goto_0
-    iget-object v14, v1, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget-object v5, v1, Lcom/mediatek/ims/ril/ImsRILAdapter;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v5}, Ljava/lang/Integer;->intValue()I
-
-    move-result v15
-
     iget v5, v12, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
 
     const/16 v17, 0x3
@@ -17828,8 +17772,6 @@
     .line 4559
     move/from16 v16, v5
 
-    invoke-virtual/range {v14 .. v20}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeRilSendSms(IIIIJ)V
-
     .line 4567
     goto :goto_2
 
diff --git a/smali/com/mediatek/wfo/ril/MwiRIL$RilHandler.smali b/smali/com/mediatek/wfo/ril/MwiRIL$RilHandler.smali
index 15fbacd..a0e2359 100644
--- a/smali/com/mediatek/wfo/ril/MwiRIL$RilHandler.smali
+++ b/smali/com/mediatek/wfo/ril/MwiRIL$RilHandler.smali
@@ -183,27 +183,6 @@
 
     invoke-virtual {v1}, Landroid/os/Message;->sendToTarget()V
 
-    .line 408
-    iget-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL$RilHandler;->this$0:Lcom/mediatek/wfo/ril/MwiRIL;
-
-    invoke-static {v1}, Lcom/mediatek/wfo/ril/MwiRIL;->-$$Nest$fgetmMetrics(Lcom/mediatek/wfo/ril/MwiRIL;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v1
-
-    iget-object v4, p0, Lcom/mediatek/wfo/ril/MwiRIL$RilHandler;->this$0:Lcom/mediatek/wfo/ril/MwiRIL;
-
-    iget-object v4, v4, Lcom/mediatek/wfo/ril/MwiRIL;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v4}, Ljava/lang/Integer;->intValue()I
-
-    move-result v4
-
-    iget v5, v2, Lcom/mediatek/wfo/ril/RILRequest;->mSerial:I
-
-    iget v6, v2, Lcom/mediatek/wfo/ril/RILRequest;->mRequest:I
-
-    invoke-virtual {v1, v4, v5, v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilTimeoutResponse(III)V
-
     .line 411
     .end local v3    # "timeoutResponse":Ljava/lang/Object;
     :cond_1
@@ -286,12 +265,8 @@
 
     invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
-    move-result-object v3
-
     invoke-virtual {v3, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
 
-    move-result-object v3
-
     invoke-virtual {v3}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
 
     move-result-object v3
diff --git a/smali/com/mediatek/wfo/ril/MwiRIL.smali b/smali/com/mediatek/wfo/ril/MwiRIL.smali
index 3552d70..83adbff 100644
--- a/smali/com/mediatek/wfo/ril/MwiRIL.smali
+++ b/smali/com/mediatek/wfo/ril/MwiRIL.smali
@@ -83,8 +83,6 @@
 
 .field mIsMobileNetworkSupported:Z
 
-.field private mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
 .field final mPhoneId:Ljava/lang/Integer;
 
 .field private mRILDefaultWorkSource:Landroid/os/WorkSource;
@@ -129,14 +127,6 @@
 
 
 # direct methods
-.method static bridge synthetic -$$Nest$fgetmMetrics(Lcom/mediatek/wfo/ril/MwiRIL;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-    .locals 0
-
-    iget-object p0, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    return-object p0
-.end method
-
 .method static bridge synthetic -$$Nest$mclearWakeLock(Lcom/mediatek/wfo/ril/MwiRIL;I)Z
     .locals 0
 
@@ -267,14 +257,7 @@
 
     iput-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mTestingEmergencyCall:Ljava/util/concurrent/atomic/AtomicBoolean;
 
-    .line 295
-    invoke-static {}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->getInstance()Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v1
-
-    iput-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    .line 307
+    .line 305
     const/4 v1, 0x0
 
     iput-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mRadioProxy:Lvendor/mediatek/hardware/mtkradioex/V3_0/IMtkRadioEx;
@@ -3436,23 +3419,6 @@
 
     .line 1175
     :goto_0
-    iget-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget-object v0, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v0}, Ljava/lang/Integer;->intValue()I
-
-    move-result v2
-
-    iget v3, p1, Lcom/mediatek/wfo/ril/RILRequest;->mSerial:I
-
-    iget v4, p2, Landroid/hardware/radio/V1_0/RadioResponseInfo;->error:I
-
-    iget v5, p1, Lcom/mediatek/wfo/ril/RILRequest;->mRequest:I
-
-    move-object v6, p3
-
-    invoke-virtual/range {v1 .. v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilSolicitedResponse(IIIILjava/lang/Object;)V
 
     .line 1177
     if-eqz p1, :cond_2
-- 
2.43.0

